### ISSUE TITLE
molarMass in PartialTwoPhaseMedium is partial

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#794]
**Reported by sjoelund&#x2e;se on 22 Aug 2012 09:06 UTC**
My apologies if I did not research this thoroughly enough, but the Media library is hard to navigate. I found this error while investigating Modelica.Media.Examples.Tests.MediaTestModels.LinearFluid.LinearWater_pT. In our flat Modelica, we got a partial function Modelica.Media.Water.StandardWater.molarMass. But this function is complete, and has a statement coming from [Modelica 3.1/Media/package.mo:5732:9-5732:42] (as far as I could tell, PartialTwoPhaseMedium). The same code seems to exist in MSL 3.2.

```
redeclare replaceable partial function extends molarMass
  "Return the molar mass of the medium"
algorithm
  MM := fluidConstants[1].molarMass;
end molarMass;
```

Since the function is never redeclared again, was it simply an error to mark it as partial and these should be omitted from the code, or am I missing something? There are more functions that were partial redeclare extended and had algorithm sections added to them.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/794

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 22 Aug 2012 09:19 UTC**
That is an error and certainly should be fixed. I would be glad to get a complete list of those functions: the compiler is better than human eyes in finding all occurrences of that problem. /Hubertus

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 22 Aug 2012 09:24 UTC**
Our compiler can only find functions that were actually called, but I will try to make a list using some grep magic.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#46;se on 22 Aug 2012 09:35 UTC**
Maybe I was wrong about the other functions; setState_XXX functions are usually redeclared, right?

```
$ grep -R "redeclare.*partial.*function.*extends" {MSL31,3.1-MediaFix,Modelica-trunk}/Modelica/Media | grep -v svn
MSL31/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_dTX
MSL31/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_phX
MSL31/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_psX
MSL31/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_pTX
MSL31/Modelica/Media/package.mo:      redeclare replaceable partial function extends molarMass
3.1-MediaFix/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_dTX
3.1-MediaFix/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_phX
3.1-MediaFix/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_psX
3.1-MediaFix/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_pTX
3.1-MediaFix/Modelica/Media/package.mo:      redeclare replaceable partial function extends molarMass
Modelica-trunk/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_dTX
Modelica-trunk/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_phX
Modelica-trunk/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_psX
Modelica-trunk/Modelica/Media/package.mo:    redeclare replaceable partial function extends setState_pTX
Modelica-trunk/Modelica/Media/package.mo:      redeclare replaceable partial function extends molarMass
```

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 22 Aug 2012 09:45 UTC**
Yes, setState_XXX functions should always be redeclared. Thanks for finding this.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 22 Aug 2012 12:08 UTC**
This was fixed in 5416-5418 in trunk and the 3.2.1 release branch.
